utils/convert.go

```diff
@@ -13,21 +13,25 @@ import (
 	"unsafe"
 )
 
+const MaxStringLen = 0x7fff0000 // Maximum string length for UnsafeBytes. (decimal: 2147418112)
+
 // #nosec G103
 // UnsafeString returns a string pointer without allocation
 func UnsafeString(b []byte) string {
 	return *(*string)(unsafe.Pointer(&b))
 }
 
 // #nosec G103
-// UnsafeBytes returns a byte pointer without allocation
-func UnsafeBytes(s string) (bs []byte) {
-	sh := (*reflect.StringHeader)(unsafe.Pointer(&s))
-	bh := (*reflect.SliceHeader)(unsafe.Pointer(&bs))
-	bh.Data = sh.Data
-	bh.Len = sh.Len
-	bh.Cap = sh.Len
-	return
+// UnsafeBytes returns a byte pointer without allocation.
+// String length shouldn't be more than 2147418112.
+func UnsafeBytes(s string) []byte {
+	if s == "" {
```

Comment:
What happens if the MaxStringLen limit is exceeded? Why not check for this in here as well?

Reply:
Will panic like `panic: runtime error: slice bounds out of range `

Reply:
okay for me, since this limit is high enough